controllers/openstackbaremetalset_controller.go

```diff
@@ -316,22 +314,40 @@ func (r *OpenStackBaremetalSetReconciler) Reconcile(ctx context.Context, req ctr
 	}
 
 	//
-	//   Create/Update IPSet for the BMSet CR
+	// get OSNetCfg object
 	//
-	ipset, ctrlResult, err := openstackipset.CreateOrUpdateIPset(
-		r,
-		instance,
-		cond,
-		common.IPSet{
-			Networks:            instance.Spec.Networks,
-			Role:                instance.Spec.RoleName,
-			HostCount:           instance.Spec.Count,
-			AddToPredictableIPs: true, // right now all BMSet are tripleo roles
-			HostNameRefs:        instance.GetHostnames(),
-		},
-	)
-	if (err != nil) || (ctrlResult != ctrl.Result{}) {
-		return ctrlResult, err
+	osnetcfg := &ospdirectorv1beta1.OpenStackNetConfig{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      strings.ToLower(instance.Labels[openstacknetconfig.OpenStackNetConfigReconcileLabel]),
+		Namespace: instance.Namespace},
+		osnetcfg)
+	if err != nil && !k8s_errors.IsNotFound(err) {
+		cond.Message = fmt.Sprintf("Failed to get %s %s ", osnetcfg.Kind, osnetcfg.Name)
+		cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.MACCondReasonError)
+		cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.CommonCondTypeError)
+		err = common.WrapErrorForObject(cond.Message, instance, err)
+
+		return ctrl.Result{}, err
+	}
+
+	//
+	// Wait for IPs created on all configured networks
+	//
+	for hostname, hostStatus := range instance.Status.BaremetalHosts {
+		err = openstacknetconfig.WaitOnIPsCreated(
+			instance,
+			cond,
+			osnetcfg,
```

Comment:
Couldn't `osnetcfg` be an empty `OpenStackNetConfig` if the `err` above was `IsNotFound`?  If so, would `WaitOnIPsCreated` behave as expected if it received the empty object?

Reply:
good catch, we should wait until osnetcfg is there. it does not make sense to proceed as we don't have networks

Reply:
ah wait AddOSNetConfigRefLabel [1] at the top of the controller is waiting for all osnet to be available. so the osnetcfg should be there. but lets just check for all osnetcfg get errors here.

[1] https://github.com/openstack-k8s-operators/osp-director-operator/blob/master/pkg/openstacknetconfig/funcs.go#L47

Reply:
changed